src/main/java/nl/knaw/dans/ingest/core/deposit/DepositReaderImpl.java

```diff
@@ -117,6 +127,58 @@ Deposit mapToDeposit(Path path, Path bagDir, Configuration config, Bag bag) {
         return deposit;
     }
 
+    private OriginalFilePathMapping getOriginalFilePathMapping(Path bagDir) throws IOException {
+        var originalFilepathsFile = bagDir.resolve("original-filepaths.txt");
+
+        if (Files.exists(originalFilepathsFile)) {
+            var lines = Files.readAllLines(originalFilepathsFile);
+            var mappings = lines.stream().map(line -> {
+                    var parts = line.split(" {2}", 2);
+
+                    if (parts.length == 2) {
+                        return new OriginalFilePathMapping.Mapping(
+                            Path.of(parts[0].trim()),
+                            Path.of(parts[1].trim())
+                        );
+                    }
+
+                    return null;
+                })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
+
+            return new OriginalFilePathMapping(mappings);
+        }
+        else {
+            return new OriginalFilePathMapping(Set.of());
+        }
+    }
+
+    private List<DepositFile> parseFileList(Bag bag, Document filesXml) throws IOException {
+        if (filesXml == null) {
+            return List.of();
+        }
+
+        var bagDir = bag.getRootDir();
+        var filePathToSha1 = ManifestHelper.getFilePathToSha1(bag);
+        var originalFilePathMappings = getOriginalFilePathMapping(bagDir);
+
+        return XPathEvaluator.nodes(filesXml, "/files:files/files:file")
+            .map(node -> {
+                var filePath = Optional.ofNullable(node.getAttributes().getNamedItem("filepath"))
+                    .map(Node::getTextContent)
+                    .map(Path::of)
+                    .orElseThrow(() -> new IllegalArgumentException("File element with filepath attribute"));
```

Comment:
I am confused. shouldn't his be "without"?

Reply:
correct